include/libtorrent/i2p_stream.hpp

```diff
@@ -277,7 +277,7 @@ struct i2p_stream : proxy_base
 				break;
 		}
 
-		string_view remaining(m_buffer.data(), m_buffer.size());
+		string_view remaining(m_buffer.data(), m_buffer.size() - 1);
```

Comment:
Fixing it this way makes me a but uneasy. Especially by not also ensuring that the character being truncated is in fact the new-line. It would seem a bit more logical to truncate the last character of `m_buffer` on line 241, instead of adding the null terminator. I believe this buffer used to be treated as null terminated.

Reply:
Ok, I will look if I can implement it better.

Reply:
Looks like null is required at line 247.
I can change `m_dest = &m_buffer[0];` to `m_dest.assign(&m_buffer[0], m_buffer.size());` there.
And then at line 241 `m_buffer[read_pos - 1] = 0;` to `m_buffer.pop_back();`
@arvidn will it be better than ` - 1`?

Reply:
that's a good point. So this is really just ignoring the null terminator in `m_buffer`.
would you mind adding a `TORRENT_ASSERT(m_buffer[m_buffer.size() - 1] == '\0');`

Reply:
I added assert in f6d329a4c0df214053edee9d2a11457e89476bfe, but slightly modified it because of compiler complaints.

Reply:
at a second thought it would probably have been nicer to make it:
```
TORRENT_ASSERT(m_buffer.back() == '\0');
```
but what you have is fine.